lib/msf/core/exploit/remote/kerberos/client.rb

```diff
@@ -155,8 +155,8 @@ def send_request_tgt(options = {})
             server_name = options[:server_name]
             client_name = options[:client_name]
             password = options[:password]
-            password.force_encoding('utf-8') if password
-            client_name.force_encoding('utf-8')
+            password.freeze.dup.force_encoding('utf-8') if password
+            client_name.freeze.dup.force_encoding('utf-8')
```

Comment:
```suggestion
            password.dup.force_encoding('utf-8') if password
            client_name.dup.force_encoding('utf-8')
```

My bad; The `.freeze` call was just an example of how `dup` would still work with frozen strings - which often happens when the frozen_string_literal magic comment is is present https://stackoverflow.com/questions/37799296/ruby-what-does-the-comment-frozen-string-literal-true-do

Reply:
Gotcha - thanks for that.